lib/babe/verify.go

```diff
@@ -344,29 +344,53 @@ func (b *verifier) verifyAuthorshipRight(header *types.Header) error {
 	// hashes is hashes of all blocks with same block number as header.Number
 	hashes := b.blockState.GetAllBlocksAtDepth(header.ParentHash)
 
-	for _, hash := range hashes {
-		currentHeader, err := b.blockState.GetHeader(hash)
+	for _, currentHash := range hashes {
+		currentHeader, err := b.blockState.GetHeader(currentHash)
 		if err != nil {
+			logger.Errorf("failed get header %s", err)
 			continue
 		}
 
 		currentBlockProducerIndex, err := getAuthorityIndex(currentHeader)
 		if err != nil {
+			logger.Errorf("failed to get authority index %s", err)
 			continue
 		}
 
+		if len(currentHeader.Digest.Types) == 0 {
+			logger.Errorf("current header missing digest")
```

Comment:
I think we should jump to the next item otherwise this condition will let empty `currentHeader.Digest.Types` go further and panic the node when tried to `currentPreDigestItem := currentHeader.Digest.Types[0]`

Reply:
I have started returning things, it makes more sense since it's a verification function.